### PR TITLE
update flashcard api and setup edgestore

### PR DESCRIPTION
**Pull request by william1nguyen**

### CODE REVIEW

File: lib/edgestore.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use client';
+
+import { type EdgeStoreRouter } from '@/app/api/edgestore/[...edgestore]/route';
+import { createEdgeStoreProvider } from '@edgestore/react';
+
+const { EdgeStoreProvider, useEdgeStore } =
+  createEdgeStoreProvider<EdgeStoreRouter>({
+    maxConcurrentUploads: 2,
```
Comment (luzivaltic) [inline lib/edgestore.ts:8]: why ?

File: app/pages/api/flashcards/route.ts [hunk 2]
```diff
@@ -59,7 +61,7 @@ export const PUT = async (req: Request) => {
   }
 
   try {
-    const { flashcardId, chapterId } = await req.json();
```
Comment (luzivaltic) [inline app/pages/api/flashcards/route.ts:62]: no chapterId

File: lib/edgestore.ts (hunk 1 above)
Comment (nolannguyen1212) [reply lib/edgestore.ts:8]: you can only upload maximum 2 files to EdgeStore at the same time.

Comment (nolannguyen1212) [reply lib/edgestore.ts:8]: just to limit the number of files loaded into EdgeStore

**Review by luzivaltic — APPROVED** [review]

**Comment by luzivaltic** [comment]
Remove the code smell issue which SonarCloud found

**Pull request merged**